.github/workflows/bbw_build_container_rhel.yml

```diff
@@ -59,44 +72,55 @@ jobs:
             echo "IMG=${TAG_TMP/:/}" >>$GITHUB_ENV
           fi
           echo "REPO=bb-worker" >>$GITHUB_ENV
+
       - name: Check for rhel subscription credentials
         if: >
-          github.repository == 'mariadb/buildbot' &&
-          github.ref == 'refs/heads/main'
+          github.repository == 'mariadb/buildbot'
         run: |
           missing=()
           [[ -n "${{ secrets.RHEL_ORGID }}" ]] || missing+=(RHEL_ORGID)
           [[ -n "${{ secrets.RHEL_KEYNAME }}" ]] || missing+=(RHEL_KEYNAME)
           for i in "${missing[@]}"; do
             echo "Missing github secret: $i"
           done
-          (( ${#missing[@]} == 0 )) || exit 1
-          echo "BUILD_RHEL=true" >> $GITHUB_ENV
+
+          if (( ${#missing[@]} == 0 )); then
+            echo "BUILD_RHEL=true" >> $GITHUB_ENV
+          else
+            echo "Not building RHEL"
+          fi
+
       - name: Generate Dockerfile and necessary files
+        if: ${{ env.MAIN_BRANCH == 'false' }}
```

Comment:
@fauust This is what I thought initially, because I was used to Azure DevOps Pipelines where you can group as many steps as you like under a single IF statement. 

This is not possible in GitHub workflows and it's a bit disappointing that I need to write that condition on every step, making it more error prone.